src/lifetime.rs

```diff
@@ -15,14 +17,18 @@ pub struct Lifetime(pub Timer);
 /// Despawn entities who's lifetime has expired
 fn lifetime_system(
     mut commands: Commands,
-    mut entities: Query<(Entity, &mut Lifetime)>,
+    mut entities: Query<(Entity, &mut Lifetime, Option<&mut Drop>)>,
     time: Res<Time>,
 ) {
-    for (entity, mut lifetime) in &mut entities {
+    for (entity, mut lifetime, drop) in &mut entities {
         lifetime.tick(time.delta());
 
         if lifetime.finished() {
-            commands.entity(entity).despawn_recursive();
+            if let Some(mut drop) = drop {
```

Comment:
I don't like having to make the lifetime system specifically tailored for the droppable items.

I think this is another argument in favor of using an event to drop items.

Maybe we make a `LifetimeExpired` event, that will be emitted by the `lifetime_system` when something is deleted by it. This keeps the lifetime plugin fully agnostic of the rest of the plugins.

Then we can have the `drop_system` listen for `LifetimeExpired`, or `ItemDropped` ( or whatever it is called ) events.

Reply:
I added events, made the `LifetimeExpired`, but I added a `Drop` and `Transform` fields to it, not sure how to reference the new code on `lifetime.rs` but could you take a look and see if it's agnostic enough? Tried making the event with just a entity field but found some complications to get the `Drop` and `Transform` on the `drop_system` with just a entity.